Adds Arm-Mods (Mounted Taser & Mounted Laser)

PKPenguin321 opened the pull request:
Adds two new implants, the Arm-cannon Taser Implant and the Arm-cannon Laser Implant. They are implanted into the chest.
They can be toggled via an action button. When on, a mounted taser/laser is placed into your hand (it appears to replace your arm) that self-charges and cannot be dropped. When off, the weapon is sucked back into your arm. The taser has two modes, tase and disable.

If you get EMPed with one of these implants, your item will forcibly be sucked back into your arm, you'll take 25 burn damage, get ignited, and the implant breaks.

Currently, these implants are admin-only (they're stupidly powerful).

Also adds a parent object that is easily modified to allow for future arm-mounted items. It's thoroughly commented and easy to follow.

Sprites by newfren.
![megaman](https://cloud.githubusercontent.com/assets/8345184/11991189/042089b8-a9cc-11e5-800f-858d853b7190.png)

:cl: PKPenguin321
rscadd: Chest implants have been added that allow you to morph your arms into a gun and back at will. There are two variants, a taser version and a laser version. Both versions are self-charging. Getting EMPed with one of these implants results in the implant breaking and loads of fire damage. The implants are currently admin-only.
/:cl:

PKPenguin321 committed:
Adds Arm-Mods.
Can be easily made with the example parent type. Includes two premade arm-mods, a mounted taser and a mounted laser.

as334 commented:
I can become megaman :+1:

optimumtact commented on code/modules/surgery/organs/augments_internal.dm:
```diff
@@ -253,6 +253,71 @@
+		owner.IgniteMob()
+		owner.say("AUUUUUUUUUUUUUUUUUUGH!!")
+		owner.adjustFireLoss(10000)//kill the owner!
+		overloaded = 1//then make sure this can't happen again by breaking the implant.
```
you may wish to place the taser back into null space in case it's actually out in the users hands

kevinz000 commented:
... Seems to me like one hit insta kill shouldn't be a thing for hte EMP.. It's just inf ammo taser/laser, that's overpowered but shouldn't warrent a 1 hit kill by an EMP.

newfren commented:
I made sprites dad are you proud of me.

KorPhaeron commented:
>If you get EMPed with one of these implants, you instantly husk and die an agonizing death

That is kind of overboard, don't you think?

Dorsisdwarf commented:
Nah I like the idea of something stupidly powerful having an even more massive weakness

Supermichael777 commented:
If we had dismemberment and this was in the arms where it really should be then it could just blow your arm off. anyway just make them explode 1,2,4 and then gib them. that death doesn't have nearly enough collateral damage

KorPhaeron commented:
A lasergun really isnt massively powerful compared to instant death because an aoe effect touched you.

And if it WAS massively powerful having a single hard counter the crew had no access to would be bad

mazen30 commented:
I am pleased that my Idea is gonna come true

PKPenguin321 commented:
EMPs are so rare, though. Strength of the counter (in the case, an EMP) should scale inversely with "Number of Counters / Availability of Counters" which in this case is a very small number.
But I'll nerf the EMP act anyways, just to be nice to you guys. New commit coming very soon.

PKPenguin321 committed:
tweaks EMP act
now deals 25 fire damage and ignites the owner with a full 20 firestacks. not an instant kill. also forcibly retracts the item should it be out at the time of EMP.

mazen30 commented:
The item shouldn't be a one hit kill item nor admin spawnable, just would need lots of resources to make or something hard to aquire, like the core needed for phazons

duncathan commented on code/modules/surgery/organs/augments_internal.dm:
```diff
@@ -253,6 +253,76 @@
+/obj/item/organ/internal/cyberimp/chest/arm_mod/ui_action_click()
+	if(overloaded == 0)//ensure the implant isn't broken
+		if(out == 1)//check if the owner has the item out already
+			owner.unEquip(holder, 1)//if he does, take it away. then,
```
`if(!overloaded)`
`if(out)`

duncathan commented on code/modules/surgery/organs/augments_internal.dm:
```diff
@@ -253,6 +253,76 @@
+	if(!owner)//ensure that it's in an owner, then...
+		return
+	if(overloaded == 0)//make sure this is the first time it's been EMPed, and...
+		if(out == 1)//check if he has the item out...
```
See above

duncathan replied:
Also for both these instances, instead of having all this behind two if checks, just make it return if overloaded is true

PKPenguin321 commented:
i want to see about making it obtainable later, for now I just want to get the item in the game

PKPenguin321 committed:
simplifies some if() statements
thanks @duncathan

PKPenguin321 commented:
This has been tested and is ready for review. I'd like if it could get merged before the inevitable conflicts.

PKPenguin321 commented:
@duncathan mergebegging

duncathan commented:
@Wjohnston any issues with the sprites?

PKPenguin321 commented:
M E R G E B E G

PKPenguin321 commented:
@duncathan WJ hasn't spoken in 10,000 years and if this gets merge conflicts because it was left unmerged I'll die
If you can't merge it right now, maybe your buddies @tkdrg or @RemieRichards can? [Mergebegging intensifies]

Jordie0608 commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -0,0 +1,28 @@
+	name = "mounted taser"
+	desc = "An arm mounted dual-mode weapon that fires electrodes and disabler shots."
+	icon_state = "armcannon"
+	force = "5"
```
No quote marks, this should be a number, not a string

Jordie0608 commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -0,0 +1,28 @@
+	desc = "An arm mounted cannon that fires lethal lasers. Doesn't come with a charge beam."
+	icon_state = "armcannon"
+	item_state = "armcannonlase"
+	force = "5"
```
Same

Jordie0608 commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -0,0 +1,28 @@
+	selfcharge = 1
+	flags = NODROP
+	slot_flags = null
+	w_class = 5.0
```
Remove the .0 please

Jordie0608 commented on code/modules/projectiles/guns/mounted.dm:
```diff
@@ -0,0 +1,28 @@
+	selfcharge = 1
+	flags = NODROP
+	slot_flags = null
+	w_class = 5.0
```
Same

Jordie0608 commented on code/modules/surgery/organs/augments_internal.dm:
```diff
@@ -253,6 +253,73 @@
+			owner.visible_message("<span class='notice'>You extend [holder]!</span>")
+			playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+		else//if this fails to put the item in his hands,
+			holder.loc = null//keep it in nullspace
```
I don't think this is needed

optimumtact replied:
it could fail if their hands are full

Jordie0608 replied:
I mean this line, is the loc changed to non-null if the check fails?

PKPenguin321 replied:
Yes, during testing the check would fail and it would place the NODROP item on the ground, which is a big no-no

Jordie0608 commented on code/modules/surgery/organs/augments_internal.dm:
```diff
@@ -253,6 +253,73 @@
+	name = "Arm-cannon laser implant"
+	desc = "A variant of the arm cannon implant that fires lethal laser beams. The cannon emerges from the subject's arms and remains in the shoulders when not in use."
+	icon_state = "armcannon_lase_implant"
+	origin_tech = "materials=5;combat=5;biotech=4;powerstorage=4;syndicate=5"//this is kinda nutty and i might lower it
```
Generally syndicate tech levels are given to things only traitors can get, it's admin only so I guess it doesn't matter too much

PKPenguin321 replied:
I'll leave it as-is for now, if/when it ever makes it in the game later on I'll lower it

PKPenguin321 commented:
Fucking jordie being all helpful and shit now I have to do work and all my mergebegging is in vain

Just kidding, thanks for doing an actually thorough review. I'll push another commit hopefully within 24 hours. When its out I will mergebeg until the maintainers die of old age though

PKPenguin321 committed:
fixes the force values and removes ".0"s

thanks @jordie0608
pls let this be ready for merge

PKPenguin321 commented:
Okay, this should be really *really* merge ready for realsies now.

PKPenguin321 commented:
This has been open for a week now and has been pretty much 100% ready to merge for a day

Pretty pretty please merge

optimumtact commented:
come in irc and merge beg there it's more effective

duncathan commented:
well I don't know much about spriting but those don't look disgusting so I guess I'll just make an executive decision

my apologies @WJohn if it turns out I'm blind

WJohn commented:
Eh, the sprites aren't great but they're passable enough. They could do with some improvements in the future, though.

PKPenguin321 commented:
Oh, NOW you speak up.

optimumtact commented:
dont' bully wjohn they are a busy person
